backend/src/graph-info/graph-info.controller.ts

```diff
@@ -0,0 +1,26 @@
+import { Controller, Get, Query } from '@nestjs/common';
+import { GraphInfoService } from './graph-info.service';
+import { EntityType } from '../shared/graph-information/EntityType';
+
+@Controller('graphInfo')
+export class GraphInfoController {
+  constructor(private readonly graphInfoService: GraphInfoService) {}
+
+  /**
+   * Returns information about all nodes or edges of a graph
+   * @param whichQuery either 'nodes' or 'edges'
+   */
+  @Get('getEntityTypes')
+  getEntityTypes(@Query('which') whichQuery: string): Promise<EntityType[]> {
```

Comment:
Split this into to methods `getNodeTypes` and `getEdgeTypes`. It makes no sense IMO to implement it this way. If you need both, node types and edge types, you have to call this twice anyway.
Also add types `NodeType` and `EdgeType` that both inherit from `EntityType`. It is not necessary to add additional members, but this makes it clear, already from the types what you are dealing with and prevents errors that you missrepresent an instance of `EntityType` for the wrong type (node/edge).